<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>bot_preferences</class>
 <widget class="QWidget" name="bot_preferences">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>701</width>
    <height>777</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="6" column="0">
    <widget class="QGroupBox" name="guardrails_groupbox">
     <property name="title">
      <string>Guardrails</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QPlainTextEdit" name="guardrails_prompt">
        <property name="placeholderText">
         <string>The guardrails prompt is used to moderate results.</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QGroupBox" name="system_instructions_groupbox">
     <property name="title">
      <string>System Instructions</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QPlainTextEdit" name="system_instructions">
        <property name="placeholderText">
         <string>Instructions for the LLM</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Chatbots</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QComboBox" name="saved_chatbots"/>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="create_new_button">
        <property name="text">
         <string>Create New</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="pushButton">
        <property name="text">
         <string>Delete</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QGroupBox" name="personality_groupbox">
     <property name="title">
      <string>Bot Personality</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="1" column="0">
       <widget class="QPlainTextEdit" name="bot_personality">
        <property name="placeholderText">
         <string>EXAMPLE: {{ botname }} is very helpful and {{ gender }} loves {{ username }}.</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>A brief description of the bot's personality</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QGroupBox" name="mood_groupbox">
     <property name="title">
      <string>Bot Mood</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QPlainTextEdit" name="bot_mood">
        <property name="placeholderText">
         <string>EXAMPLE: Excited</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox" name="names_groupbox">
     <property name="title">
      <string>Use names</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>User name</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="username">
          <property name="text">
           <string>User</string>
          </property>
          <property name="placeholderText">
           <string>User name</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Assistant name</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="botname">
          <property name="text">
           <string>AI Runner</string>
          </property>
          <property name="cursorPosition">
           <number>9</number>
          </property>
          <property name="placeholderText">
           <string>Bot name</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>username</sender>
   <signal>textChanged(QString)</signal>
   <receiver>bot_preferences</receiver>
   <slot>username_changed(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>206</y>
    </hint>
    <hint type="destinationlabel">
     <x>55</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>botname</sender>
   <signal>textChanged(QString)</signal>
   <receiver>bot_preferences</receiver>
   <slot>botname_changed(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>252</y>
    </hint>
    <hint type="destinationlabel">
     <x>127</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bot_personality</sender>
   <signal>textChanged()</signal>
   <receiver>bot_preferences</receiver>
   <slot>bot_personality_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>338</x>
     <y>355</y>
    </hint>
    <hint type="destinationlabel">
     <x>292</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bot_mood</sender>
   <signal>textChanged()</signal>
   <receiver>bot_preferences</receiver>
   <slot>bot_mood_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>494</x>
     <y>503</y>
    </hint>
    <hint type="destinationlabel">
     <x>451</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>names_groupbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>bot_preferences</receiver>
   <slot>toggle_use_names(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>42</x>
     <y>213</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>personality_groupbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>bot_preferences</receiver>
   <slot>toggle_use_personality(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>90</x>
     <y>286</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>guardrails_prompt</sender>
   <signal>textChanged()</signal>
   <receiver>bot_preferences</receiver>
   <slot>guardrails_prompt_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>160</x>
     <y>718</y>
    </hint>
    <hint type="destinationlabel">
     <x>1</x>
     <y>379</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>system_instructions</sender>
   <signal>textChanged()</signal>
   <receiver>bot_preferences</receiver>
   <slot>system_instructions_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>306</x>
     <y>610</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>545</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mood_groupbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>bot_preferences</receiver>
   <slot>toggle_use_mood(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>486</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>276</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>guardrails_groupbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>bot_preferences</receiver>
   <slot>toggle_use_guardrails(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>65</x>
     <y>711</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>510</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>system_instructions_groupbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>bot_preferences</receiver>
   <slot>toggle_use_system_instructions(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>578</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>424</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saved_chatbots</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>bot_preferences</receiver>
   <slot>saved_chatbots_changed(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>235</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>2</x>
     <y>27</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>create_new_button</sender>
   <signal>clicked()</signal>
   <receiver>bot_preferences</receiver>
   <slot>create_new_chatbot_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>446</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>7</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>bot_preferences</receiver>
   <slot>delete_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>515</x>
     <y>110</y>
    </hint>
    <hint type="destinationlabel">
     <x>480</x>
     <y>-13</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>username_changed(QString)</slot>
  <slot>botname_changed(QString)</slot>
  <slot>bot_personality_changed()</slot>
  <slot>bot_mood_changed()</slot>
  <slot>toggle_use_names(bool)</slot>
  <slot>toggle_use_personality(bool)</slot>
  <slot>toggle_use_mood(bool)</slot>
  <slot>toggle_use_guardrails(bool)</slot>
  <slot>guardrails_prompt_changed()</slot>
  <slot>reset_guardrails_to_default()</slot>
  <slot>system_instructions_changed()</slot>
  <slot>toggle_use_system_instructions(bool)</slot>
  <slot>saved_chatbots_changed(QString)</slot>
  <slot>create_new_chatbot_clicked()</slot>
  <slot>delete_clicked()</slot>
 </slots>
</ui>
